### PR TITLE
extend add_Design_files command with -work and -l arguments.

### PR DESCRIPTION
**Pull request by TarasSydor**
Given PR extends 'add_Design_files ' TCL command with '-work' and '-L' arguments.
It also makes sure they are passed to Yosys script. So far only Verific parser is supported, as default Yosys one puts all the files from the script into one pile.
UI is also not included here.

By default, empty strings will be used and previous behavior kept.

To make its behavior more clear, here is the example: Input (test.tcl)
...
add_design_file -work lib1 test.v -SV_2012
add_design_file -L lib1 -L lib2 bottom.v -SV_2012
...

Output (test.ys):
...
verific -work lib1  -sv2012 /home/work/Work/FOEDAG-Taras/tests/Testcases/trivial/test.v
verific -L lib1 -L lib2  -sv2012 /home/work/Work/FOEDAG-Taras/tests/Testcases/trivial/bottom.v
...

Tested on FOEDAG application with following cmdline commands:
--verific --compiler openfpga --script /home/work/Work/FOEDAG-Taras/tests/Testcases/trivial/test.tcl

### CODE REVIEW

**Comment by alaindargelas** [comment]
", as default Yosys one puts all the files from the script into one pile." Create another PR for this, as I explained, Yosys supports multiple calls to read_verilog but does not support -work or -L (Warning needed)

File: src/NewProject/ProjectManager/project_fileset.h [hunk 1]
```diff
@@ -30,12 +31,17 @@ class ProjectFileSet : public ProjectOption {
   const std::vector<std::pair<QString, QString>> &getMapFiles() const;
   const std::vector<std::pair<int, QStringList>> &Files() const;
 
+  std::vector<std::pair<QStringList, QStringList>> getLibraries() const;
```
Comment (KochynVolodymyr) [inline src/NewProject/ProjectManager/project_fileset.h:34]: const reference here

File: src/Compiler/Compiler.cpp [hunk 2]
```diff
@@ -354,6 +354,7 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
     if (argc < 2) {
       compiler->ErrorMessage(
           "Incorrect syntax for add_design_file <file(s)> "
+          "[-work libraryName] [-L libraryName1 -L libraryName2]"
```
Comment (KochynVolodymyr) [inline src/Compiler/Compiler.cpp:357]: Need update help as well. For Compiler and CompilerOpenFPGA

**Comment by TarasSydor** [comment]
> ", as default Yosys one puts all the files from the script into one pile." Create another PR for this, as I explained, Yosys supports multiple calls to read_verilog but does not support -work or -L (Warning needed)

Will do. I didn't include it here because -work and -L are optional and won't change the default behavior, while Yosys parser change will affect everyone.

**Comment by alain-rs** [comment]
@TarasSydor please update the PR, @ravic-rs please merge when ready.
I need this change to move forward with https://github.com/os-fpga/FOEDAG/pull/601

**Pull request merged**